yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,396 @@
+            path[0] = keep;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(uniswap).swapExactTokensForTokens(
+                keepBalance,
+                0,
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Deposit acquired WBTC to the Saddle pool to gain additional
+        // vault's underlying tokens.
+        uint256 wbtcBalance = IERC20(wbtc).balanceOf(address(this));
+        if (wbtcBalance > 0) {
+            IERC20(wbtc).safeIncreaseAllowance(tbtcSaddlePoolSwap, wbtcBalance);
+
+            uint256[] memory amounts = new uint256[](4);
+            amounts[1] = wbtcBalance;
+
+            bytes32[] memory merkleProof = new bytes32[](1);
+
+            //TODO: Check what the values of minToMint, deadline and merkleProof
+            // should be
+            ISaddlePoolSwap(tbtcSaddlePoolSwap).addLiquidity(
+                amounts,
+                0,
+                uint256(-1),
+                merkleProof
+            );
+        }
+
+        // Check the profit and loss in the context of strategy debt.
+        uint256 totalAssets = estimatedTotalAssets();
+        uint256 totalDebt = vault.strategies(address(this)).totalDebt;
+        if (totalAssets < totalDebt) {
+            loss = totalDebt - totalAssets;
+            profit = 0;
+        } else {
+            profit = balanceOfWant().sub(initialWantBalance);
+        }
+
+        // Repay some vault debt if needed.
+        if (debtOutstanding > 0) {
+            withdrawSome(debtOutstanding);
+            debtPayment = Math.min(
+                debtOutstanding,
+                balanceOfWant().sub(profit)
+            );
+        }
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to define all tokens/tokenized positions this contract manages
+    ///         on a persistent basis (e.g. not just for swapping back to
+    ///         the want token ephemerally).
+    /// @dev Should not include want token, already included in the base contract.
+    /// @return Addresses of protected tokens
+    function protectedTokens()
+        internal
+        view
+        override
+        returns (address[] memory)
+    {
+        address[] memory protected = new address[](1);
+        protected[0] = keep;
+        return protected;
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to provide an accurate conversion from amtInWei (denominated in wei)
+    ///         to want token (using the native decimal characteristics of want token).
+    /// @param amtInWei The amount (in wei/1e-18 ETH) to convert to want tokens.
+    /// @return The amount in want tokens.
+    function ethToWant(uint256 amtInWei)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        // TODO: Create an accurate price oracle.
```

Comment:
We can implement this just like it was proposed in https://github.com/keep-network/tbtc-v2/pull/15 using Saddle's `calculateTokenAmount` method.

Reply:
Done at https://github.com/keep-network/tbtc-v2/pull/13/commits/1d94f9943e965f45638341f1f2d941636caf3d53.